### PR TITLE
fix circle.yml

### PR DESCRIPTION
**Pull request by vrancurel**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,3 +1,15 @@
+dependencies:
+  post:
+    - sudo apt-get update
+    - sudo apt-get -y install clang-3.9 libargtable2-dev libjsoncpp-dev doxygen libmicrohttpd-dev
+    - git clone git://github.com/cinemast/libjson-rpc-cpp.git
+    - cd libjson-rpc-cpp && git checkout v0.6.0
+    - mkdir libjson-rpc-cpp/build
+    - cd libjson-rpc-cpp/build && cmake -DCOMPILE_TESTS=No .. && make && sudo make install
+    - sudo ldconfig
+    - cd src ; make
```
Comment (slaperche-scality) [inline circle.yml]: This should be in a compile section, no?

According to https://circleci.com/docs/1.0/configuration/:
> compile: compile your project

Sth like

```yaml
compile:
  override:
    - cd src; make
```

File: lint.rb [hunk 2]
```diff
@@ -36,5 +36,5 @@
 ].freeze
 TO_CHECK = Dir.glob('src/*.{cpp,h}') - BLACKLIST
 
-exec("clang-tidy #{TO_CHECK.join(' ')} "\
+exec("clang-tidy-3.9 #{TO_CHECK.join(' ')} "\
```
Comment (slaperche-scality) [inline lint.rb]: We should get rid of the version number here by creating a symlink in the dependencies.

File: lint.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env ruby
 # encoding: utf-8
 
-COMPIL_OPTS = '-x c++ --std=c++11'.freeze
+COMPIL_OPTS = '-x c++ --std=c++11 -DHAVE_READLINE'.freeze
```
Comment (slaperche-scality) [inline lint.rb]: Is this change needed?

I think HAVE_READLINE is optional (if set we use GNU Readline, if unset we fallback on a lighter version: anyway, it should have no impact on the linting).

File: src/Makefile [hunk 4]
```diff
@@ -1,10 +1,10 @@
-CC = clang++
+CC = clang++-3.9
 
 CFLAGS = \
--std=c++11 -g -Werror -Wall -Wextra -Wdeprecated -Wpedantic \
```
Comment (slaperche-scality) [inline src/Makefile]: Ha yeah, my clang is probably too recent.

That being said, this issue should disappear with my PR to use CMake (where we only add flags that are supported by the compiler currently in use) :)

File: src/Makefile [hunk 5]
```diff
@@ -1,10 +1,10 @@
-CC = clang++
+CC = clang++-3.9
```
Comment (slaperche-scality) [inline src/Makefile]: I prefer that we avoid pinning clang version (for instance I use clang 5.0), IMHO this should be fixed by creating a symlink in the `dependencies` step of the circle.yml file.
Like what's done [here](https://github.com/status-im/react-native-desktop/blob/react-native-qt/.circleci/config.yml#L21).

File: circle.yml [hunk 6]
```diff
@@ -1,3 +1,15 @@
...
+    - sudo apt-get -y install clang-tidy-3.9 
+
 test:
   override:
-    - ./lint.rb
+    - ./lint.rb | grep -v error:
```
Comment (slaperche-scality) [inline circle.yml]: For the check to be effective (failing if there is any linter error), we should probably grep on `warning:`.

Something like that:

```sh
test `./lint.rb | grep -c warning:` -eq 0
```

Comment (vrancurel) [reply circle.yml]: how do you test warning and errors ?

**Comment by vrancurel** [comment]
updated

File: circle.yml (hunk 6 above)
Comment (slaperche-scality) [reply circle.yml]: With something like

```sh
test `./lint.rb | grep -Pc (warning|error):` -eq 0
```

It should do the trick. But I think that lint error are only warnings.
Error are compilation error (and thus should be catched during the compilation step), no?
